ModpackUploader.ps1

```diff
@@ -1,54 +1,91 @@
 . .\settings.ps1
 
-function Download-GithubRelease {	
-    param(	
-        [parameter(Mandatory=$true)]	
-        [string]	
-        $repo,	
+function Download-GithubRelease {
+    param(
         [parameter(Mandatory=$true)]
-        [string]	
-        $file	
-    )	
+        [string]
+        $repo,
+        [parameter(Mandatory=$true)]
+        [string]
+        $file
+    )
+
+    $releases = "https://api.github.com/repos/$repo/releases"
 
-    $releases = "https://api.github.com/repos/$repo/releases"	
+    Write-Host "Determining latest release of $repo"
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    $tag = (Invoke-WebRequest -Uri $releases -UseBasicParsing | ConvertFrom-Json)[0].tag_name
 
-    Write-Host "Determining latest release of $repo"	
-    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12	
-    $tag = (Invoke-WebRequest -Uri $releases -UseBasicParsing | ConvertFrom-Json)[0].tag_name	
+    $download = "https://github.com/$repo/releases/download/$tag/$file"
 
-    $download = "https://github.com/$repo/releases/download/$tag/$file"	
-    $name = $file.Split(".")[0]	
+    if ($IsWindows) {
+        $name = $file.Split(".")[0]
+    }
 
-    Write-Host Dowloading...	
-    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12	
-    Invoke-WebRequest $download -Out $file	
+    Write-Host Dowloading...
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    Invoke-WebRequest $download -Out $file
 
-    # Cleaning up target dir	
-    Remove-Item $name -Recurse -Force -ErrorAction SilentlyContinue	
-}	
+    # Cleaning up target dir
+    if ($IsWindows) {
+        Remove-Item $name -Recurse -Force -ErrorAction SilentlyContinue
+    }
+}
 
 function Clear-SleepHost {
     Start-Sleep 2
     Clear-Host
 }
-
-if (-not (test-path "$env:ProgramFiles\7-Zip\7z.exe")) {throw "$env:ProgramFiles\7-Zip\7z.exe needed to use the ModpackUploader."} 
-    Set-Alias sz "$env:ProgramFiles\7-Zip\7z.exe"
-
-if (-not (test-path "$env:C:\Windows\System32\curl.exe")) {throw "$env:Windows\System32\curl.exe needed to use the ModpackUploader."} 
-    Set-Alias cl "$env:C:\Windows\System32\curl.exe"
+if ($IsLinux) {
+    #Lets Check if the user has 7-Zip Installed
+    if (-not (test-path "/usr/bin/7z")) {Write-Host "7-Zip needed to use the ModpackUploader."}
```

Comment:
I don't really get the logic here:
- You check if 7zip exists
- Print that it's needed if it doesn't exist
- then set it's alias ???
- check if it exists again
- then break
why not just check -> print -> break if it doesn't exist, and set alias if it does?

Reply:
Resolved over Discord.